src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -189,10 +208,12 @@ void CliffordSynthesizer::depthOptimalSynthesis(
     // minimizing over the layers of gates (=timesteps) in the resulting
     // circuit.
     runMaxSAT(config);
+  } else if (configuration.linearSearch) {
+    runLinearSearch(config.timestepLimit, lower, upper, config);
   } else {
     // The binary search approach calls the SAT solver repeatedly with varying
-    // timestep (=depth) limits T until a solution with depth T is found, but no
-    // solution with depth T-1 could be determined.
+    // timestep (=gate) limits T until a solution with T gates is found, but no
+    // solution with T-1 gates could be determined.
```

Comment:
this change seems wrong to me.

Reply:
Not entirely sure what I was thinking there...